### DISCUSSION TITLE
How to mute or bypass nodes w/o an output?

### DISCUSSION THREAD

**Discussion by aegis72** [rgthree/rgthree-comfy#116] [Q&A] [answered]
Occasionally, especially when dealing with nodes that save things or show a string or whatever, I can mute or bypass everything UP TO the final node. But because the bypasser takes OUTPUTS, rather than (or additionally) having a node that executes the bypass forward, there's often a node that is last in the chain that can't be bypassed:

![image](https://github.com/rgthree/rgthree-comfy/assets/118572301/e8045abc-9cd3-4a8f-ad2f-2295330a5027)

The WF still runs, but throws errors in console and redboxes the node, which is confusing to users. How might I mute the last node in this (or really any) chain?

**Comment by rgthree** [comment, marked as answer]
Unfortunately, for the original Fast Muter/Bypasser the node must have an output to connect it to. In your example it's actually OK that the Image Save node isn't muted because you've muted the nodes that connect into it.

Like in your screenshot, it will turn red because it thinks there's an error, but as long as you have another output node (Image Preview, Image Save, etc) it won't actually stop the execution; just be red which is fine.

If you want to mute everything, including nodes w/o outputs, you can put them in a Group and use the Fast Groups Muter or Fast Groups Bypasser, which will have toggles for each Group and mute/bypass/enable everything in the group at once.

**Reply by aegis72** [reply]
Thanks; because of my use-case, group bypassing won't work in this case. I will just tell folks what I often do...red doesn't mean dead.

I suppose I could hack those nodes to give them an output, even a fake one would work...but that is a logistical nightmare I don't want :)
